crypto/x509/x509_vfy.c

```diff
@@ -509,6 +509,12 @@ static int check_chain_extensions(X509_STORE_CTX *ctx)
                 ret = 1;
             break;
         }
+        if ((x->ex_flags & EXFLAG_CA) == 0
+            && x->ex_pathlen != -1
+            && ctx->param->flags & X509_V_FLAG_X509_STRICT) {
+            ctx->error = X509_V_ERR_INVALID_EXTENSION;
```

Comment:
do we need a verify callback here?

Reply:
(I dont know in the moment) the answer is open.

Reply:
Perhaps, though applications would be wise to not ignore such errors, especially non-leaf certificates, since we can't provide much context here... For now, we can leave this the way it is (no callback).